src/main/webapp/utils.ts

```diff
@@ -49,46 +53,252 @@ export function fallOnSameDay(time1: number, time2: number): boolean {
-    let date = new Date(0);
-    // Convert key from string to number
-    date.setTime(+key);
-    let tooltip = getTooltip(+key, value, data.dateToRecommendationTaken);
-    rows.push([date, value, tooltip])
+  let out: Date[] = [];
+  days.forEach(time => {
+    out.push(new Date(time));
+  });
+
+  out.sort((a, b) => a.getTime() - b.getTime());
+  return out;
+}
+
+/** Creates the table rows from the given ProjectGraphData The first row contains the column headers */
+export function createIamRows(graphData: ProjectGraphData[], colors?: string[], days?: Date[]): any[] {
+  if (!days) {
+    // First, get all the days we need to add if it hasn't already been provided
+    days = uniqueDays(graphData);
   }
+  if (!colors) {
+    colors = defaultColors;
+  }
+
+  // Each row is [time, data1, data1-tooltip, data1-style, data2, data2-tooltip, ...]
+  let rows: any[] = [];
+  // Add a row for each unique day
+  days.forEach(day => rows.push([day]));
+
+  graphData.forEach((data, index) => {
+    for (const [key, value] of Object.entries(data.dateToNumberIAMBindings)) {
+      // Convert key from string to number
+      let date = startOfDay(+key);
+      // The row we're adding to is the same index as unique days
+      let row = rows[days.findIndex(findDate => findDate.getTime() === date.getTime())];
 
-  return rows as [[Date, number, string]];
+      let recommendations = getRecommendations(+key, data.dateToRecommendationTaken);
+      let tooltip = getTooltip(value, recommendations);
+      let point = getPoint(recommendations, colors[index]);
+
+      // Populate the existing row with information
+      row.push(value, tooltip, point);
+    }
+  });
+
+  return rows;
+}
+
+/** Creates the column headers for an IAM graph */
+export function createIamColumns(graphData: ProjectGraphData[]): any[] {
+  let columns: any[] = ['Time'];
+  graphData.forEach(data => {
+    // Populate the header row, which contains the column purposes
+    columns.push(data.projectId, { type: 'string', role: 'tooltip' }, { type: 'string', role: 'style' });
+  });
+  return columns;
+}
+
+export function createIamOptions(graphData: ProjectGraphData[], colors?: string[]): google.visualization.LineChartOptions {
+  let options: google.visualization.LineChartOptions = {
+    animation: {
+      duration: 250,
+      easing: 'ease-in-out',
+      startup: true
+    },
+    legend: { position: 'none' },
+    height: 700,
+    width: 1000,
+    hAxis: {
+      gridlines: {
+        color: 'white'
+      }
+    },
+    vAxis: {
+      minorGridlines: {
+        color: 'white'
+      }
+    },
+    series: {}
+  }
+
+  if (!colors) {
+    colors = defaultColors;
+  }
+  graphData.forEach((data, index) => {
+    options.series[index] = { color: colors[index % colors.length] };
+  })
+  return options;
+}
+
+/** Pull the colors from projects into the order of projects in graphData so colors can be assigned easily */
```

Comment:
It is a very poor way of managing colors, yes. The other branch (which I realize I'm bringing up far too often) has the color logic simplified considerably. Again, I'm inclined to leave it be in this branch to avoid duplicating work I've already done.
Clearly I need to rethink how I'm doing things, since I keep changing things for the better in the other branch.